### PR TITLE
Introduce feature_flag for `credhub`

### PR DESCRIPTION
**Pull request by qu1queee**
For [issue 399](https://github.com/SUSE/kubecf/issues/399)

Add conditionals in all YAMLs files where we set spec properties for `credhub`, based
on the feature flag conditional(`.Values.features.credhub.enabled`).

Added  overrides for the base `cf-deployment.yml` file, based on the current v12.18.0 version.

## Description
Adding the feature flag conditionals in all related places where we define spec properties for `credhub`. Also, because `credhub` is by default defined in the `cf-deployment.yml`, we need to remove some spec properties also there.

## Motivation and Context
The reasoning is that in IBM, we do not use the `credhub` component. Also, I will expect in the future more use cases, where `credhub` requires to be `feature_flag`.

## How Has This Been Tested?
I deploy this current PR branch in an IKS cluster two times, the first with the feature_flag disable, the second one with the feature_flag enable. Both deployments were successful, and I was able to push applications.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by f0rmiga** [comment]
@qu1queee Thanks for the PR! Could you resolve the conflict, please?

File: deploy/helm/kubecf/templates/sizing.yaml [hunk 1]
```diff
@@ -82,6 +82,8 @@ data:
       value: 1
 {{- end }}
 
+
+{{- if .Values.features.credhub.enabled -}}
```
Comment (f0rmiga) [inline deploy/helm/kubecf/templates/sizing.yaml]:
```suggestion
{{- if .Values.features.credhub.enabled }}
```

**Comment by qu1queee** [comment]
Working on this, I was on holidays during the end of last week,

**Comment by f0rmiga** [comment]
I had an issue with credhub disabled and CATS enabled. It's fairly simple to fix. I'll push a commit to fix it here. Other than that, it worked fine! Thanks again.

**Pull request merged**